src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt

```diff
@@ -399,6 +399,21 @@ class SentencingService(
     }
   }
 
+  // use offender charge to reset all CEC common values
+  private fun refreshCourtEventCharges(
+    offenderCharge: OffenderCharge,
+    case: CourtCase,
+  ) {
+    case.courtEvents.flatMap { it.courtEventCharges }.filter { it.id.offenderCharge == offenderCharge }
+      .forEach { courtEventCharge ->
+        courtEventCharge.offenceDate = offenderCharge.offenceDate
+        courtEventCharge.offenceEndDate = offenderCharge.offenceEndDate
+        courtEventCharge.mostSeriousFlag = offenderCharge.mostSeriousFlag
+        courtEventCharge.resultCode1 = offenderCharge.resultCode1
```

Comment:
Actually are you sure about the result code? Wouldn't that mean if the outcome on the final appearance was "Imprisonment" wouldn't it then update the first appearance result to "imprisonment" .
It doesn't "feel" right. It does for offenceDate etc but not the result

Reply:
will raise it with DPS as I don't think they have the ability to have different charge data for each appearance